core/src/block_sync.rs

```diff
@@ -191,10 +191,11 @@ pub mod message {
                     previous_hash,
                     peer_id,
                 }) => {
-                    if block_sync.block_batch_size == 0 {
-                        warn!("Error: not sending any blocks as batch_size is equal to zero.");
-                        return;
-                    }
+                    // FIXME: is it okay to remove this behaviour?
+                    // if block_sync.block_batch_size == 0 {
+                    //     warn!("Error: not sending any blocks as batch_size is equal to zero.");
+                    //     return;
+                    // }
```

Comment:
@Erigara Have you introduced this? I believe we can remove this

Reply:
I don't remember, but setting block batch size to 0 can break consensus.

Because observing peers receive blocks only from block sync.